src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1078,36 +1077,270 @@ protected void workerDone(Worker worker) {
   }
   
   /**
-   * <p>Runnable which will consume tasks from the appropriate and given the provided 
-   * implementation to get a worker and execute consumed tasks.</p>
+   * <p>A service which manages the execute queues.  It runs a task to consume from the queues and 
+   * execute those tasks as workers become available.  It also manages the queues as tasks are 
+   * added, removed, or rescheduled.</p>
+   * 
+   * <p>Right now this class has a pretty tight dependency on {@link PriorityScheduler}, which is 
+   * why this is an inner class.</p>
    * 
    * @author jent - Mike Jensen
-   * @since 1.0.0
+   * @since 3.4.0
    */
-  protected class TaskConsumer extends BlockingQueueConsumer<TaskWrapper> {
```

Comment:
Its probably worth doing but I dont know that its worth doing now.  I think we might want to go over this code a little more I have a feeling we could get around some of the locking we are having to do here.